### ISSUE TITLE
emit warning for properties with just a getter that are not set in a constructor

### ISSUE THREAD

**Issue by arnovb-github** [dotnet/roslyn-analyzers#3137]
When a property has only a getter, no implementation, its contents can only be set by a constructor. But constructors are optional. Without a constructor setting that property, those properties just default to whatever their return type. It would be nice if the debugger could emit a warning on such properties.

**Comment by arnovb-github** [comment]
@jaredpar yes that is what I meant. Thank you for the time and effort you spent on my suggestion. I'll elaborate a bit: sometimes the compiler takes care of so many things for you behind the scenes that it actually makes the language harder to understand. Imho, this is one of those cases. Unless you know about constructors, realizing that a property with only an implicit getter *can* actually be set can be mind-blowing esp. to novices.

**Comment by Evangelink** [comment]
I am not so sure this is a good idea, especially if the goal is only for the clarity that a readonly property can be _set_. Regarding the fact that a user might forget to initialize a property's default value then I think this is the exact same scenario as for fields so if an analyzer is added it shall cover both fields and properties.

**Comment by arnovb-github** [comment]
@Evangelink  it may not a good idea, I simply don't know. I am glad to leave it up to the considerable brainpower of the community. What I would like to elaborate on some more is language clearness. You can create a implicit get-only property. The only way to set it is via a constructor, but constructors are also implicit. At this point, it is my perception that there is so much implicitness going on that it hurts the clarity of the language, hence my suggestion.

**Comment by Evangelink** [comment]
Giving some more thoughts about this, I actually think that's worth having some rule warning on non-explicitly initialized readonly properties and fields. As they are readonly it makes sense to have a clear intent by actually going for an explicit initialization even for `null`s. Anyways I'll let the repo owners be judge here.

**Comment by Evangelink** [comment]
Tagging @mavasani
